spec/utils/request_verifier_spec.rb

```diff
@@ -0,0 +1,140 @@
+                expect { @request_verifier_object.verify_request_signature(request_body, request_header, signing_secret, tolerance) }.to raise_error("Request time exceeded tolerance threshold. Request: 1900871395" + ", CurrentTime: " + currentTime.to_s + ", tolerance: " + tolerance.to_s)
+            end
+        end
+    end
+
+    describe '#verifySignature' do
+        context 'Signature request is unverified, signing secret does not exist in signatures, potential typo' do
+            it 'throws "Unverified signature request, If this request was unexpected, it may be from a bad actor. Please proceed with caution. If the request was exepected, please check any typos or issues with the signingSecret"' do
+                currentTime = DateTime.now.strftime('%s').to_i
+                request_header = "t=1679944186,v1=c3957749baf61df4b1506802579cc69a74c77a1ae21447b930e5a704f9ec4120,v1=1ba18712726898fbbe48cd862dd096a709f7ad761a5bab14bda9ac24d963a6a8"
+                signing_secret = "sigsec_ead6d3b6904196c60835d039e91b3341c77a7794"
+                tolerance = (5 * 60)
+                request_body =  "{\"accountId\":\"AC1334ffb694cd8d969f51cddf5f7c9b478546d50c\",\"callId\":\"CAccb0b00506553cda09b51c5477f672a49e0b2213\",\"callStatus\":\"ringing\",\"conferenceId\":null,\"direction\":\"inbound\",\"from\":\"+13121000109\",\"parentCallId\":null,\"queueId\":null,\"requestType\":\"inboundCall\",\"to\":\"+13121000096\"}"
+                expect { @request_verifier_object.verify_request_signature(request_body, request_header, signing_secret, tolerance) }.to raise_error("Unverified signature request, If this request was unexpected, it may be from a bad actor. Please proceed with caution. If the request was exepected, please check any typos or issues with the signingSecret")
+            end
+        end
+    end
+
+    describe '#verifyRequestSignature' do
+        context 'Request is valid' do
+            it 'No errors are thrown' do
+                currentTime = DateTime.now.strftime('%s').to_i
+                request_header = "t=1679944186,v1=c3957749baf61df4b1506802579cc69a74c77a1ae21447b930e5a704f9ec4120,v1=1ba18712726898fbbe48cd862dd096a709f7ad761a5bab14bda9ac24d963a6a8"
+                signing_secret = "sigsec_ead6d3b6904196c60835d039e91b3341c77a7793"
+                tolerance = (5 * 60)
+                request_body =  "{\"accountId\":\"AC1334ffb694cd8d969f51cddf5f7c9b478546d50c\",\"callId\":\"CAccb0b00506553cda09b51c5477f672a49e0b2213\",\"callStatus\":\"ringing\",\"conferenceId\":null,\"direction\":\"inbound\",\"from\":\"+13121000109\",\"parentCallId\":null,\"queueId\":null,\"requestType\":\"inboundCall\",\"to\":\"+13121000096\"}"
+                expect { @request_verifier_object.verify_request_signature(request_body, request_header, signing_secret, tolerance) }.not_to raise_error
+            end
+        end
+    end
+    
+end
```

Comment:
rename all methods in descriptions to correspond with name updates

Reply:
This will be changed in latest commit